libdnf/conf/config_main.cpp

```diff
@@ -359,6 +361,19 @@ ConfigMain::Impl::Impl(Config & owner) : owner(owner) {
     owner.opt_binds().add("pluginpath", pluginpath);
     owner.opt_binds().add("pluginconfpath", pluginconfpath);
     owner.opt_binds().add("persistdir", persistdir);
+
+    // Unless transaction_history_dir has been explicitly set, use the system_state_dir as its default
+    owner.opt_binds().add(
+        "system_state_dir",
+        system_state_dir,
+        [&](Option::Priority, const std::string & value) {
+            transaction_history_dir.set(Option::Priority::DEFAULT, value);
```

Comment:
I am not sure if `Option::Priority::DEFAULT` is correct. @jrohel Please can you take a look?

Reply:
So `system_state_dir` sets the default value of `transaction_history_dir`. So how do we set the value of `system_state_dir`? A line of code may be missing.

```
        [&](Option::Priority priority, const std::string & value) {
            system_state_dir.set(priority, value);
            transaction_history_dir.set(Option::Priority::DEFAULT, value);
        },
```

Reply:
Right, thanks for noticing, fixed.